adagrams/game.py

```diff
@@ -1,11 +1,118 @@
+import random
+import string
 def draw_letters():
-    pass
+    hand_of_letters = []
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+    
+    #iterate through the dict to convert letter_pool to a list
+    LETTER_POOL_list = []  
+    for letter,count in LETTER_POOL.items():
+        while count > 0:
+            LETTER_POOL_list.append(letter)
+            count -=1    
+   
+    total_letter_count = len(LETTER_POOL_list)
+    hand_of_letters = []
+    #to get 10 letters 
+    for i in range(10):
+        #draw a random index from 0 to (totoal_letter_cout - i), given there are i letters
+        #there are already drawn & put back as the last i items in the list
+        draw_index = random.randrange(0, total_letter_count - i)
+        hand_of_letters.append(LETTER_POOL_list[draw_index])
+        # move the drawed letter to the end and won't be picked in next round
+        #swith two items in the list : a, b = b,a
+        LETTER_POOL_list[draw_index], LETTER_POOL_list[total_letter_count - i - 1] = \
+            LETTER_POOL_list[total_letter_count - i - 1], LETTER_POOL_list[draw_index]
```

Comment:
When you have a long line of code like this that needs to be broken up, you might want to consider what can you do to make it shorter. Having long lines of code like this can reduce readability. 

Using variables with descriptive names will make your code a little longer which is ok because it will make your code easier to understand.

```python
randomly_drawn_letter = LETTER_POOL_list[draw_index]
last_letter = LETTER_POOL_list[total_letter_count - i - 1] 

randomly_drawn_letter, last_letter = last_letter, randomly_drawn_letter
```

Reply:
Thank you!!